### ISSUE TITLE
Mismatch between reported and reproduced retrieval results for ViT-L-14 aligned checkpoint

### ISSUE THREAD

**Issue by showstarpro** [peterant330/KUEA#7]
Hi, thank you for releasing the code and checkpoints for KUEA.

We evaluated the provided [ViT-L-14 aligned checkpoint](https://mycuhk-my.sharepoint.com/:f:/g/personal/1155187960_link_cuhk_edu_hk/Evj3UUqXLpRNjwQ0pQi-NugB7-JuKxU4xxGiqjrBH_MRDA?e=7SNAec)and found that the zero-shot classification results are almost identical to those reported in the paper, but the image-text retrieval results on COCO and Flickr30K show noticeable discrepancies.

For zero-shot evaluation, the reproduced result is consistent with the paper:

ViT-L-14 kernel average zero-shot
Reported in Table 1: 66.54
Our reproduced result: 66.56

So the checkpoint itself seems to be generally correct.

However, for image-text retrieval, our reproduced results differ from those reported in Table 2:

**COCO**

Image-to-Text R@1
Reported: 51.72
Ours: 50.98
Text-to-Image R@1
Reported: 35.98
Ours: 35.74

**Flickr30K**

Image-to-Text R@1
Reported: 83.00
Ours: 79.20
Text-to-Image R@1
Reported: 64.78
Ours: 60.70

The discrepancy on COCO is moderate, but the gap on Flickr30K is relatively large.

Since the zero-shot results are well aligned with the paper, we are wondering whether there might be some differences in the retrieval evaluation protocol, such as:

dataset split / annotation version
image preprocessing or text tokenization
evaluation script details
checkpoint selection
whether any additional fine-tuning or special setting was used for retrieval evaluation

We are attaching our retrieval evaluation logs below for reference.

Could you please help clarify the exact evaluation setup used for Table 2, especially for COCO and Flickr30K retrieval?

Thanks a lot for your help.

## Evaluation Logs

{"dataset": "wds/mscoco_captions", "model": "ViT-L-14", "pretrained": "/models/showstarpro__kuea/26-04-16-0248/ViT-L-14-openai-kuea.pt", "beta": null, "task": "zeroshot_retrieval", "metrics": {"image_retrieval_recall@1": 0.3574170470237732, "text_retrieval_recall@1": 0.5098000168800354, "image_retrieval_recall@5": 0.6071571111679077, "text_retrieval_recall@5": 0.751800000667572, "image_retrieval_recall@10": 0.7073570489883423, "text_retrieval_recall@10": 0.8366000056266785}, "language": "en", "attack": "none", "iterations_adv": 100, "eps": 1.0, "norm": "Linf"}

{"dataset": "wds/flickr30k", "model": "ViT-L-14", "pretrained": "/models/showstarpro__kuea/26-04-16-0248/ViT-L-14-openai-kuea.pt", "beta": null, "task": "zeroshot_retrieval", "metrics": {"image_retrieval_recall@1": 0.6069999933242798, "text_retrieval_recall@1": 0.7919999957084656, "image_retrieval_recall@5": 0.8456000089645386, "text_retrieval_recall@5": 0.953000009059906, "image_retrieval_recall@10": 0.8999999761581421, "text_retrieval_recall@10": 0.9789999723434448}, "language": "en", "attack": "none", "iterations_adv": 100, "eps": 1.0, "norm": "Linf"}

**Comment by peterant330** [comment]
We did not conduct any addtional fine-tuning or checkpoint selection when performing the retrieval. It is purely zero-shot. I uploaded the dataset I used [here](https://mycuhk-my.sharepoint.com/:f:/g/personal/1155187960_link_cuhk_edu_hk/IgBGhbNb24CaSYs3dtxOfzuKAZHl5FoJlFxnxBohzHthQHI?e=yliVuP). You may have a try. You may also want to evaluate the performance of the original CLIP with the same setup. I think it would be fine as long as their results are aligned.

**Comment by showstarpro** [comment]
OK，I use the dataset from the huggingface [[clip-benchmark/wds_mscoco_captions](https://huggingface.co/datasets/clip-benchmark/wds_mscoco_captions)] and [wds_fer2013](https://huggingface.co/datasets/clip-benchmark/wds_fer2013). The performance of the original OpenAI checkpoint also degraded. I think this might be caused by other factors. However, Kuea's method is still effective.

<img width="2042" height="296" alt="Image" src="https://github.com/user-attachments/assets/fc2e89f9-fc98-42e3-9700-d20fd9a51ab7" />

<img width="2068" height="290" alt="Image" src="https://github.com/user-attachments/assets/19ce3851-a863-4817-b77d-ce274c337801" />

**Comment by showstarpro** [comment]
When we attempted to reproduce using this codebase with ViT-L-14, ImageNet-1K, 2 epochs, and a batch size of 128×8, we modified part of the step calculation code as follows:
```
steps_per_epoch = len(dataloader)
args.steps = args.total_epochs * steps_per_epoch
```

The total number of steps is 40000/16 = 2500. The warmup was also adjusted proportionally, scaled from 2800 down to 175. The full args are shown below:
`Namespace(clip_model_name='ViT-L-14', pretrained='openai', dataset='imagenet', template='std', imagenet_root='/lpai/dataset/imagenet-1k/0-1-0', imagenet21k_root='/mnt/datasets/imagenet', vision_model='dino', output_normalize=False, start_step=0, optimizer_state='', total_epochs=2, steps=40000, warmup=175, batch_size=128, loss='l2', loss_clean='l2', penalty_weight=0.5, clean_weight=1.0, band_clip=8.7, band_dino=15.8, gamma=0.0032, coef0=0.191623, kernel_dino='polynomial', kernel_clip='polynomial', trades=False, opt='adamw', momentum_sgd=0.9, lr=1e-05, wd=0.0001, inner_loss='l2', wandb=False, experiment_name='exp_1', overwrite=False, log_freq=1, eval_freq=10, output_dir='/lpai/output/models/ViT-L-14_openai_imagenet_l2_imagenet_exp_1_0UAqF', save_checkpoints=True, devices='', wds_path='', wds_train_length=2800000, finetuned_model_name='ViT-L-14_openai_imagenet_l2_imagenet_exp_1_0UAqF', rank=0, local_rank=0, world_size=8, device=device(type='cuda', index=0))`
After evaluation, we found that the model showed no improvement on either zero-shot classification or image-text retrieval.

<img width="2016" height="288" alt="Image" src="https://github.com/user-attachments/assets/778cd239-ca74-401b-9a3d-088dd062c131" />

<img width="2016" height="404" alt="Image" src="https://github.com/user-attachments/assets/ba190fb6-ebb6-46a2-b9fa-79e8205d799f" />

<img width="2244" height="378" alt="Image" src="https://github.com/user-attachments/assets/32ee8776-02dd-4585-a3f3-8822815badd7" />

Based on your training experience, is there anything I might have done incorrectly? Thank you for your response, and we hope you can provide some guidance on this matter. [Our checkpoint](https://drive.google.com/file/d/135Xle19mv1u6MnsH-LUJbbxUlxa283v5/view?usp=sharing)

**Comment by peterant330** [comment]
I think one possibility is that you utilize a much greater batch size, the number of optimization steps are only 1/16 that of my previous setting. Therefore, you should use a larger learning rate and training for longer epochs. I am not sure what are the best learning rate and training epochs for such setting. As a first step, I would recommend you to reproduce the results using the same hyperparameters as my previous experiments.
